tasks/run.yml

```diff
@@ -51,22 +51,60 @@
       {% if helm_kubectl_context is defined and helm_kubectl_context %}
       --kube-context {{ helm_kubectl_context }}
       {% endif %}
+      {% if ansible_check_mode %}
+      --dry-run
+      {% endif %}
       --version {{ item.chart_version }}
       --namespace {{ item.namespace }}
       --values {{ helm_configuration_files }}/{{ item.values_file_path }}
       --name {{ item.release }}
   with_items: "{{ helm_charts }}"
   when: item.release not in helm_release_names
+  check_mode: False
+  register: _helm_install_chart
+
+- name: Show dry mode results
+  debug:
+    msg:
+      cmd: "{{ item.cmd | join(' ') }}"
+      stdout: "{{ item.stdout }}"
+      stderr: "{{ item.stderr }}"
+      return_code: "{{ item.rc }}"
+  with_items: "{{ _helm_install_chart.results }}"
+  when:
+    # Run only in --check mode
+    - ansible_check_mode
+    # Run only for items which were not skipped
+    - (item.skipped is not defined) or (not item.skipped)
 
 - name: Update chart (if release exist already)
   command: |
     helm upgrade
       {% if helm_kubectl_context is defined and helm_kubectl_context %}
       --kube-context {{ helm_kubectl_context }}
       {% endif %}
+      {% if ansible_check_mode %}
+      --dry-run
```

Comment:
I guess this will then be always reported as changed or?

Reply:
Yes. Unfortunately I couldn't find a solution to keep this unchanged.

Reply:
Refactored